Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Projection/Feature/NodeRemoval.php

```diff
@@ -83,6 +88,155 @@ protected function removeRelationRecursivelyFromDatabaseIncludingNonReferencedNo
+                dimensionspacepoint,
+                dimensionspacepointhash,
+                parentnodeanchor,
+                childnodeanchor
+            )
+            SELECT name, position, :contentStreamIdentifier AS contentstreamidentifier,
+                dimensionspacepoint, dimensionspacepointhash, parentnodeanchor, childnodeanchor FROM (
+                    /**
+                     * This provides a list of all hierarchy relations to be copied:
+                     * parentnodeanchor and childnodeanchor, name and position only, the rest will be changed
+                     */
+                    WITH RECURSIVE descendantNodes(
+                        relationanchorpoint, parentnodeanchor, name, position, childnodeanchor
+                    ) AS (
+                        /**
+                         * Initial query: find all outgoing tethered child node relations
+                         * from the starting node in its origin
+                         */
+                        SELECT
+                            n.relationanchorpoint,
+                            h.parentnodeanchor,
+                            h.name,
+                            h.position,
+                            h.childnodeanchor
+                        FROM neos_contentgraph_node n
+                             JOIN neos_contentgraph_hierarchyrelation h ON n.relationanchorpoint = h.childnodeanchor
+                        WHERE h.parentnodeanchor = :relationAnchorPoint
+                          AND h.contentstreamidentifier = :contentStreamIdentifier
+                          AND h.dimensionspacepointhash = :originDimensionSpacePointHash
+                          ' . ($event->recursive ? '' : ' AND n.classification = :classification ') . '
+
+                        UNION ALL
+                            /**
+                             * Iteration query: find all outgoing tethered child node relations
+                             * from the parent node in its origin
+                             */
+                            SELECT
+                                c.relationanchorpoint,
+                                h.parentnodeanchor,
+                                h.name,
+                                h.position,
+                                h.childnodeanchor
+                            FROM
+                                descendantNodes p
+                                    JOIN neos_contentgraph_hierarchyrelation h
+                                        ON h.parentnodeanchor = p.relationanchorpoint
+                                    JOIN neos_contentgraph_node c ON c.relationanchorpoint = h.childnodeanchor
+                            WHERE h.contentstreamidentifier = :contentStreamIdentifier
+                              AND h.dimensionspacepointhash = :originDimensionSpacePointHash
+                              ' . ($event->recursive ? '' : ' AND c.classification = :classification ') . '
```

Comment:
can we add a comment to the Event description, that "recursive"=false means "with tethered nodes? (makes total sense to me; but is a bit non-obvious for "recursive=false".

... or another suggestion: Let's create a "Recursive Enum" (however we call it); with options "recursive" and "onlyTetheredDescendants"?